scm/github.go

```diff
@@ -689,6 +689,108 @@ func (s *GithubSCM) GetUserScopes(ctx context.Context) *Authorization {
 	return &Authorization{Scopes: gitScopes}
 }
 
+// CreateIssue implements the SCM interface
+func (s *GithubSCM) CreateIssue(ctx context.Context, opt *CreateIssueOptions) (*Issue, error) {
+	if !opt.valid() {
+		return nil, ErrMissingFields{
+			Method:  "CreateIssue",
+			Message: fmt.Sprintf("%+v", opt),
+		}
+	}
+	newIssue := &github.IssueRequest{
+		Title:     &opt.Title,
+		Body:      &opt.Body,
+		Assignee:  opt.Assignee,
+		Assignees: opt.Assignees,
+	}
+
+	s.logger.Debugf("CreateRepository: creating issue %s on %s Repository", opt.Title, opt.Repository)
+	issue, _, err := s.client.Issues.Create(ctx, opt.Organization, opt.Repository, newIssue)
+	if err != nil {
+		return nil, ErrFailedSCM{
+			Method:   "CreateIssue",
+			Message:  fmt.Sprintf("failed to create Issue %s, make sure all the details are correct ", opt.Title),
+			GitError: err,
+		}
+	}
+	s.logger.Debugf("CreateRepository: done creating issue %s", opt.Title)
+
+	return toIssue(issue), nil
+}
+
+// GetRepoIssue implements the SCM interface
+func (s *GithubSCM) GetRepoIssue(ctx context.Context, opt *IssueOptions) (*Issue, error) {
+	if !opt.valid() {
+		return nil, ErrMissingFields{
+			Method:  "GetRepoIssue",
+			Message: fmt.Sprintf("%+v", opt),
+		}
+	}
+
+	issue, _, err := s.client.Issues.Get(ctx, opt.Organization, opt.Repository, opt.IssueNumber)
+	if err != nil {
+		return nil, ErrFailedSCM{
+			Method:   "GetRepoIssue",
+			Message:  fmt.Sprintf("Failed to get Issue %v, make sure all the details are correct ", opt.IssueNumber),
+			GitError: err,
+		}
+	}
+	return toIssue(issue), nil
+}
+
+// GetRepoIssues implements the SCM interface
+func (s *GithubSCM) GetRepoIssues(ctx context.Context, opt *IssueOptions) ([]*Issue, error) {
+	if !opt.valid() {
+		return nil, ErrMissingFields{
+			Method:  "GetRepoIssues",
+			Message: fmt.Sprintf("%+v", opt),
+		}
+	}
+	issueList, _, err := s.client.Issues.ListByRepo(ctx, opt.Organization, opt.Repository, &github.IssueListByRepoOptions{})
+	if err != nil {
+		return nil, ErrFailedSCM{
+			Method:   "GetRepoIssues",
+			Message:  fmt.Sprintf("Failed to get Issues from repo %s make sure all the details are correct ", opt.Repository),
+			GitError: err,
+		}
+	}
+	var issues []*Issue
+	for _, issue := range issueList {
+		issues = append(issues, toIssue(issue))
+	}
+
+	return issues, nil
+}
+
+// EditRepoIssue implements the SCM interface
+func (s *GithubSCM) EditRepoIssue(ctx context.Context, opt *IssueOptions, isu *CreateIssueOptions) (*Issue, error) {
```

Comment:
Similarly to `GetRepoIssue` and `GetRepoIssues` you can remove `opt *IssueOptions` here and rather use `issueNumber int`.